source/MaterialXRender/Mesh.cpp

```diff
@@ -312,4 +323,116 @@ void MeshStream::transform(const Matrix44& matrix)
     }
 }
 
+MeshStreamPtr Mesh::getStream(const string& name, bool allowFallback)
+{
+    for (const auto& stream : _streams)
+    {
+        if (stream->getName() == name)
+        {
+            return stream;
+        }
+    }
+    if (allowFallback)
```

Comment:
I'm not sure about this design choice, where the formerly const method `Mesh::getStream` will silently create new streams when data is missing.

As an alternative design, what if we were to leave `Mesh::getStream` as a const method, and move the stream construction logic into a new `Mesh::createStream` method.  In the two locations where we need fallbacks for missing streams, we would simply check for a null result from `Mesh::getStream` and then call `Mesh::createStream` explicitly.